jaxkern/approximations/rff.py

```diff
@@ -0,0 +1,35 @@
+from ..base import AbstractKernel
+from ..computations import BasisFunctionComputation
+from jax.random import KeyArray
+from typing import Dict, Any
+
+
+class RFF(AbstractKernel):
+    def __init__(self, base_kernel: AbstractKernel, num_basis_fns: int) -> None:
+        self._check_valid_base_kernel(base_kernel)
+        self.base_kernel = base_kernel
+        self.num_basis_fns = num_basis_fns
+        # Set the computation engine to be basis function computation engine
+        self.compute_engine = BasisFunctionComputation
+        # Inform the compute engine of the number of basis functions
+        self.compute_engine.num_basis_fns = num_basis_fns
+
+    def init_params(self, key: KeyArray) -> Dict:
+        base_params = self.base_kernel.init_params(key)
+        n_dims = base_params["lengthscale"].shape[0]
+        frequencies = self.base_kernel.spectral_density.sample(
+            seed=key, sample_shape=(self.num_basis_fns, n_dims)
+        )
+        base_params["frequencies"] = frequencies
+        return base_params
+
+    def __call__(self, *args: Any, **kwds: Any) -> Any:
+        pass
```

Comment:
Perhaps stupid q,  so call is never called by the user really? It just feeds into cross_covariance?

Reply:
In this case, no. In general, `__call__` is the evaluation of $k(x_i, x_j)$.